packages/calling-stateful-client/src/StreamUtils.ts

```diff
@@ -30,210 +31,214 @@ export type CreateViewResult = {
-
-  if (!renderInfo) {
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_NOT_FOUND, streamLogInfo);
-    console.warn('RemoteVideoStream not found in state');
-    return;
-  }
-
-  if (renderInfo.status === 'Rendered') {
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_ALREADY_RENDERED, streamLogInfo);
-    console.warn('RemoteVideoStream is already rendered');
-    return;
-  }
-
-  if (renderInfo.status === 'Rendering') {
-    // Do not log to console here as this is a very common situation due to UI rerenders while
-    // the video rendering is in progress.
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_RENDERING, streamLogInfo);
-    return;
-  }
+  // we can only have 3 types of createView
+  let createViewType: 'local' | 'remote' | 'unparented';
 
-  // "Stopping" only happens if the stream was in "rendering" but `disposeView` was called.
-  // Now that `createView` has been re-called, we can flip the state back to "rendering".
-  if (renderInfo.status === 'Stopping') {
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_STREAM_STOPPING, streamLogInfo);
-    internalContext.setRemoteRenderInfo(
-      callId,
-      participantKey,
-      streamId,
-      renderInfo.stream,
-      'Rendering',
-      renderInfo.renderer
-    );
-    return;
-  }
-
-  const renderer = new VideoStreamRenderer(renderInfo.stream);
-
-  internalContext.setRemoteRenderInfo(callId, participantKey, streamId, renderInfo.stream, 'Rendering', undefined);
+  // we will reuse these for local as well but we need to make sure the remote stream is passed in like before.
+  let streamId;
+  let streamType;
 
-  let view;
-  try {
-    view = await renderer.createView(options);
-  } catch (e) {
-    _logCreateRemoteStreamEvent(EventNames.CREATE_REMOTE_STREAM_FAIL, streamLogInfo);
-    internalContext.setRemoteRenderInfo(callId, participantKey, streamId, renderInfo.stream, 'NotRendered', undefined);
-    throw e;
+  if (participantId) {
+    createViewType = 'remote';
+  } else if (callId) {
+    createViewType = 'local';
+  } else {
+    // TODO update for when unparented view.
+    createViewType = 'local';
   }
 
-  // Since render could take some time, we need to check if the stream is still valid and if we received a signal to
-  // stop rendering.
-  const refreshedRenderInfo = internalContext.getRemoteRenderInfoForParticipant(callId, participantKey, streamId);
-  if (!refreshedRenderInfo) {
-    // RenderInfo was removed. This should not happen unless stream was removed from the call so dispose the renderer
-    // and clean up state.
-    _logCreateRemoteStreamEvent(EventNames.REMOTE_RENDER_INFO_NOT_FOUND, streamLogInfo);
-    renderer.dispose();
-    context.setRemoteVideoStreamRendererView(callId, participantKey, streamId, undefined);
-    return;
+  if (stream) {
+    streamType = stream.mediaStreamType;
+    if (createViewType === 'remote') {
```

Comment:
local should also have an id

Reply:
(useful for logs)

Reply:
![image](https://user-images.githubusercontent.com/2684369/219800828-6dc2a192-829a-4ecb-b2a8-61bc55a739dc.png)